frontends/p4/evaluator/evaluator.h

```diff
@@ -113,10 +114,12 @@ class Evaluator final : public Inspector, public IHasBlock {
 /// high-level constructs.
 class EvaluatorPass final : public PassManager, public IHasBlock {
     P4::Evaluator *evaluator;
+    ReferenceMap selfRefMap;
```

Comment:
Out of curiosity, why not hold the local map in a `unique_ptr`? It seems like quite big struct to have it created in all cases and the access is always through pointer anyway.